tests/subsys/bluetooth/fast_pair/storage/factory_reset/src/main.c

```diff
@@ -0,0 +1,476 @@
+	zassert_ok(err, "Settings have not been fully deleted");
+
+	err = settings_load();
+	zassert_ok(err, "Unexpected error in settings load");
+}
+
+static int settings_area_erase(void)
+{
+	int err;
+	const struct flash_area *flash_area;
+
+	err = flash_area_open(FIXED_PARTITION_ID(storage_partition), &flash_area);
+	if (err) {
+		return err;
+	}
+
+	err = flash_area_erase(flash_area, 0, FIXED_PARTITION_SIZE(storage_partition));
+	flash_area_close(flash_area);
+	if (err) {
+		return err;
+	}
+
+	return 0;
+}
+
+static void teardown_fn(void)
+{
+	int err;
+
+	err = settings_area_erase();
+	zassert_ok(err, "Unable to erase settings");
+
+	fp_ram_clear();
+}
+
+static void personalized_name_store(const char *name)
+{
+	int err;
+
+	err = fp_storage_pn_save(name);
+	zassert_ok(err, "Unexpected error in name save");
+}
+
+static void personalized_name_validate_loaded(const char *name)
+{
+	int err;
+	int ret;
+	char read_name[FP_STORAGE_PN_BUF_LEN];
+
+	err = fp_storage_pn_get(read_name);
+	zassert_ok(err, "Unexpected error in name get");
+	ret = strcmp(read_name, name);
+	zassert_equal(ret, 0, "Invalid Personalized Name");
+}
+
+static void storage_validate_empty(void)
+{
+	int err;
+	int ret;
+	char read_name[FP_STORAGE_PN_BUF_LEN];
+	int read_count;
+	struct fp_account_key account_key_list[ACCOUNT_KEY_MAX_CNT];
+
+	err = fp_data_nvs_validate_empty();
+	zassert_ok(err, "Non-volatile Fast Pair data storage is not empty after reset");
+
+	read_count = fp_storage_ak_count();
+	zassert_equal(read_count, 0, "Invalid Account Key count");
+
+	err = fp_storage_ak_get(account_key_list, &read_count);
+	zassert_ok(err, "Unexpected error in Account Key get");
+	zassert_equal(read_count, 0, "Invalid Account Key count");
+
+	err = fp_storage_pn_get(read_name);
+	zassert_ok(err, "Unexpected error in name get");
+	ret = strcmp(read_name, "");
+	zassert_equal(ret, 0, "Expected saved name to be empty");
+}
+
+static void storage_self_test_run(void)
+{
+	static const uint8_t seed = 5;
+	static const uint8_t key_count = ACCOUNT_KEY_MAX_CNT;
+	static const char *name = "storage_self_test_run";
+	int err;
+
+	storage_validate_empty();
+
+	cu_account_keys_generate_and_store(seed, key_count);
+	personalized_name_store(name);
+
+	cu_account_keys_validate_loaded(seed, key_count);
+	personalized_name_validate_loaded(name);
+
+	fp_ram_clear();
+	err = settings_load();
+	zassert_ok(err, "Unexpected error in settings load");
+
+	cu_account_keys_validate_loaded(seed, key_count);
+	personalized_name_validate_loaded(name);
+}
+
+static void test_store(void)
+{
+	storage_self_test_run();
+
+	mark_test_as_finished(RESET_TEST_FINISHED_STORE_BIT_POS);
+}
+
+static void store_data_and_factory_reset(void)
+{
+	static const uint8_t seed = 50;
+	static const uint8_t key_count = ACCOUNT_KEY_MAX_CNT;
+	static const char *name = "store_data_and_factory_reset";
+	int err;
+
+	cu_account_keys_generate_and_store(seed, key_count);
+	personalized_name_store(name);
```

Comment:
Consider checking if proper values were stored here.

Reply:
It's already covered in `test_store` test.

Reply:
Actually I think it would not hurt to make sure that data is actually stored before the factory reset (that would be more robust). Keep in mind that state of the settings is not reverted to initial state after each test.